libkbfs/md_journal.go

```diff
@@ -196,13 +196,15 @@ func makeMDJournalWithIDJournal(
 	}
 
 	if earliest != nil {
-		if earliest.BID() != latest.BID() {
+		if earliest.BID() != latest.BID() &&
+			!(earliest.BID() == NullBranchID &&
+				latest.BID() == LocalSquashBranchID) {
 			return nil, fmt.Errorf(
 				"earliest.BID=%s != latest.BID=%s",
 				earliest.BID(), latest.BID())
 		}
-		log.CDebugf(nil, "Initializing with branch ID %s", earliest.BID())
-		journal.branchID = earliest.BID()
+		log.CDebugf(nil, "Initializing with branch ID %s", latest.BID())
```

Comment:
Just noticed this nil context. Perhaps plumb through ctx too, or change to `Debug`?